lib/graphql/fragment_cache/object_helpers.rb

```diff
@@ -25,6 +25,11 @@ def raw_value(obj)
       def cache_fragment(object_to_cache = NO_OBJECT, **options, &block)
         raise ArgumentError, "Block or argument must be provided" unless block_given? || object_to_cache != NO_OBJECT
 
+        if options.key?(:if) || options.key?(:unless)
+          enabled = options.key?(:if) ? options.delete(:if) : !options.delete(:unless)
+          return block_given? ? block.call : object_to_cache if !enabled
```

Comment:
ternary combined with if look a bit hard to read, would you mind splitting them?
```suggestion
          return nil unless enabled
          return block_given? ? block.call : object_to_cache
```

Reply:
The code suggestion of returning nil doesn't work and breaks tests, because the code below should still be executed. I reshuffled the code to split the condition and the ternary.